pkg/controller/node/nodecontroller.go

```diff
@@ -561,7 +561,9 @@ func (nc *NodeController) Run() {
 						} else {
 							glog.V(4).Info("Made sure that Node %v has no %v Taint", value.Value, oppositeTaint)
 						}
-						return true, 0
+
+						// remove the node from the queue after the adding/removal of taint is successful
+						return nc.zoneNotReadyOrUnreachableTainer[utilnode.GetZoneKey(node)].Remove(node.Name), 0
```

Comment:
I don't understand why we should remove the node from the list. At line 540, we add either "notready" or "unreachable" to the node. Then, we remove the other one that we didn't add at 540, here. So, the node has one of the two taints. I think we should still keep the node in the list. Am I missing something?